Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -44,6 +44,115 @@ static wstring __GetUserDefaultLocaleName() {
 }
 
 /**
+ * Private helper that creates an uppercase copy of a CFString
+ */
+CFStringRef __CFStringCreateUppercaseCopy(CFStringRef string, CFLocaleRef locale) {
+    CFMutableStringRef ret = CFStringCreateMutableCopy(nullptr, CFStringGetLength(string), string);
+    CFStringUppercase(ret, locale);
+    return ret;
+}
+
+/**
+ * Private helper that creates the IDWriteFontList for a given font family name
+ */
+HRESULT __DWriteGetFontListForFamilyName(CFStringRef familyName, IDWriteFontList** outFontList) {
+    ComPtr<IDWriteFactory> dwriteFactory;
+    RETURN_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
```

Comment:
"In most cases it is recommended you use the shared factory object, because it allows multiple components that use DirectWrite to share internal DirectWrite state data, and thereby reduce memory usage. "

By this it means, DWRITE_FACTORY_TYPE_SHARED, which we are already doing.

---
In reply to: [87902601](https://github.com/Microsoft/WinObjC/pull/1366#discussion_r87902601) [](ancestors = 87902601,87901335,87900429)